From 09a5255395af9ce1a7b3cc41ef337e5ac072278b Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:01:10 -0400
Subject: [fs] btrfs: switch to btrfs_previous_extent_item()

Message-id: <1407179109-13714-184-git-send-email-sandeen@redhat.com>
Patchwork-id: 87112
O-Subject: [RHEL7.1 PATCH 183/422] Btrfs: switch to btrfs_previous_extent_item()
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Wang Shilong <wangsl.fnst@cn.fujitsu.com>

commit 850a8cdffe41abec9e3319d7801c49eced0778a1
Author: Wang Shilong <wangsl.fnst@cn.fujitsu.com>
Date:   Thu Feb 6 20:02:29 2014 +0800

    Btrfs: switch to btrfs_previous_extent_item()

    Since we have introduced btrfs_previous_extent_item() to search previous
    extent item, just switch into it.

    Signed-off-by: Wang Shilong <wangsl.fnst@cn.fujitsu.com>
    Reviewed-by: Filipe Manana <fdmanana@gmail.com>
    Signed-off-by: Josef Bacik <jbacik@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/backref.c b/fs/btrfs/backref.c
index 903fe68..a88da72 100644
--- a/fs/btrfs/backref.c
+++ b/fs/btrfs/backref.c
@@ -1325,38 +1325,13 @@ int extent_from_logical(struct btrfs_fs_info *fs_info, u64 logical,
  if (ret < 0)
   return ret;
 
- while (1) {
-  u32 nritems;
-  if (path->slots[0] == 0) {
-   btrfs_set_path_blocking(path);
-   ret = btrfs_prev_leaf(fs_info->extent_root, path);
-   if (ret != 0) {
-    if (ret > 0) {
-     pr_debug("logical %llu is not within "
-       "any extent\n", logical);
-     ret = -ENOENT;
-    }
-    return ret;
-   }
-  } else {
-   path->slots[0]--;
-  }
-  nritems = btrfs_header_nritems(path->nodes[0]);
-  if (nritems == 0) {
-   pr_debug("logical %llu is not within any extent\n",
-     logical);
-   return -ENOENT;
-  }
-  if (path->slots[0] == nritems)
-   path->slots[0]--;
-
-  btrfs_item_key_to_cpu(path->nodes[0], found_key,
-          path->slots[0]);
-  if (found_key->type == BTRFS_EXTENT_ITEM_KEY ||
-      found_key->type == BTRFS_METADATA_ITEM_KEY)
-   break;
+ ret = btrfs_previous_extent_item(fs_info->extent_root, path, 0);
+ if (ret) {
+  if (ret > 0)
+   ret = -ENOENT;
+  return ret;
  }
-
+ btrfs_item_key_to_cpu(path->nodes[0], found_key, path->slots[0]);
  if (found_key->type == BTRFS_METADATA_ITEM_KEY)
   size = fs_info->extent_root->leafsize;
  else if (found_key->type == BTRFS_EXTENT_ITEM_KEY)
-- 
1.7.1